Sources/_SwiftSyntaxTestSupport/SourceEditsTestUtils.swift

```diff
@@ -0,0 +1,98 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import XCTest
+
+/// Get `ConcurrentEdits` in source whose edited zones are marked with markers
+/// Also extract the markers from source to get original source and edited source
+///
+/// `⏩️` is  *start marker*, `⏸️` is *separate marker*, `⏪️` is *end marker*
+/// Contents between `⏩️` and `⏸️` are source text that before modification, contents
+/// betwwen `⏸️` and `⏪️` are source text that after modification
+/// i.e. `⏩️foo⏸️bar⏪️`, the original source is `foo` and the edited source is `bar`
+public func getEditsAndSources(_ source: String) -> (edits: ConcurrentEdits, orignialSource: Substring, editedSource: Substring) {
+  var editedSource = source.prefix(0)
+  var originalSource = source.prefix(0)
+  var concurrentEdits: [SourceEdit] = []
+
+  var lastStartIndex = source.startIndex
+  while let startIndex = source[lastStartIndex...].firstIndex(where: { $0.isStartMarker }),
+    let separateIndex = source[startIndex...].firstIndex(where: { $0.isSeperateMarker }),
+    let endIndex = source[separateIndex...].firstIndex(where: { $0.isEndMarker })
```

Comment:
Out of curiosity: Why did you choose to move `isStartMarker` etc into an extension of `Character` instead of just inlining that check here? Just seems like an unnecessary abstraction layer to me if we aren’t calling the computed properties anywhere else.

Reply:
I thought that maybe these computed properties would be called elsewhere then. But it's just an unnecessary abstraction layer now. Already changed that.